src/win/util.c

```diff
@@ -1779,19 +1779,20 @@ int uv_os_uname(uv_utsname_t* buffer) {
   return r;
 }
 
-int uv_gettimeofday(uv_timeval_t* tv) {
-  /* Based on https://doxygen.postgresql.org/gettimeofday_8c_source.html */
-  const uint64_t epoch = (uint64_t) 116444736000000000ULL;
+int64_t uv_gettimeofday() {
+  /* Difference between Windows epoch 1601-01-01_00:00:00 (UTC)          *
+   * and POSIX epoch 1970-01-01_00:00:00 in 100-nanoseconds intervals    *
+   * Based on https://doxygen.postgresql.org/gettimeofday_8c_source.html */
+  const uint64_t epoch_diff = 116444736000000000ULL;
   FILETIME file_time;
-  ULARGE_INTEGER ularge;
-
-  if (tv == NULL)
-    return UV_EINVAL;
+  uint64_t ret;
 
   GetSystemTimeAsFileTime(&file_time);
-  ularge.LowPart = file_time.dwLowDateTime;
-  ularge.HighPart = file_time.dwHighDateTime;
-  tv->tv_sec = (long) ((ularge.QuadPart - epoch) / 10000000L);
-  tv->tv_usec = (long) (((ularge.QuadPart - epoch) % 10000000L) / 10);
-  return 0;
+  ret = file_time.dwHighDateTime;
+  assert(file_time.dwHighDateTime < LONG_MAX);
+  ret <<= sizeof(file_time.dwHighDateTime) * CHAR_BIT;  /* 4 * 8 = 32 */
+  ret += file_time.dwHighDateTime;
```

Comment:
`ret += file_time.dwLowDateTime`?

Reply:
```suggestion
  ret += file_time.dwHighDateTime;
```
Ack (I keep thinking of a better way to test my code...)